### ISSUE TITLE
Chunked reading lacks a __setattr__ override for point attributes

### ISSUE THREAD

**Issue by ntw-au** [laspy/laspy#205]
When reading a file in chunked mode (`file.chunk_iterator(n)`), changes made by setting attributes such as `classification` are lost and are not written to output files.

This is because when streaming using chunked mode, `chunk_iterator` returns a `ScaleAwarePointRecord`, which implements `__setattr__` as delegating to `super()` for non-geometry fields. The superclass in this case is `PackedPointRecord` that does _not_ have a `__setattr__` override.

This issue does not affect LAS manipulation when working with the whole file at once, e.g. `file.read()`, as the appropriate `__setattr__` exists.

Workaround: call the item setter instead of attribute setter, e.g. `points['classification'] = ...` instead of `points.classification = ...`.

Minimal example (input file: [input.zip](https://github.com/laspy/laspy/files/8258071/input.zip)):

```python
import laspy
import numpy as np

# Whole-file reading
# File contents: all classifications are 4
# Expected: all classifications are 5
# Actual: all classifications are 5 ✅
with laspy.open('input.las') as infile:
    points = infile.read()
    c = np.array(points.classification)
    c += 1
    points.classification = c
    points.write('output_whole.las')

# Chunked reading
# File contents: all classifications are 4
# Expected: all classifications are 5
# Actual: all classifications are 4 ❌
with laspy.open('input.las') as infile:
    with laspy.open('output_chunked.las', mode='w', header=infile.header) as outfile:
        for points in infile.chunk_iterator(8192):
            c = np.array(points.classification)
            c += 1
            points.classification = c
            outfile.write_points(points)
```

**Comment by tmontaigu** [comment]
Should be fixed on master, will release a 2.1.3 version with the fix soon.

Thanks for the report

**Comment by ntw-au** [comment]
That's brilliant—thanks very much for your prompt attention and fix.
